File: frontend/src/types/agent.ts
export interface AgentIdentity {
  id: string
  name: string
  display_name: string
  type: 'investigation' | 'diagnosis' | 'recovery' | 'orchestrator'
  icon?: string
}

export interface ToolCallRecord {
  tool: string
  params: Record<string, unknown>
  result?: unknown
  status: 'pending' | 'running' | 'success' | 'error'
  timestamp: string
}

export interface AgentExecution {
  id: string
  session_id: string
  agent: AgentIdentity
  status: 'idle' | 'running' | 'completed' | 'failed'
  thoughts: string[]
  tool_calls: ToolCallRecord[]
  result?: string
  started_at: string
  completed_at?: string
  duration_ms?: number
}

export interface ServerMessage {
  type: string
  agent?: AgentIdentity
  payload: unknown
  timestamp: string
  sequence: number
}

export interface TimelineNode {
  id: string
  type: 'alert' | 'investigation' | 'diagnosis' | 'recovery' | 'complete'
  status: 'pending' | 'running' | 'completed' | 'failed'
  title: string
  description?: string
  timestamp: string
  duration?: number
  agent?: AgentIdentity
}

File: backend/app/models/agent.py
from datetime import datetime
from typing import Any, List, Literal, Optional

from pydantic import BaseModel


class AgentIdentity(BaseModel):
    id: str
    name: str
    display_name: str
    type: Literal["investigation", "diagnosis", "recovery", "orchestrator"]
    icon: Optional[str] = None


class ToolCallRecord(BaseModel):
    tool: str
    params: dict
    result: Optional[Any] = None
    status: Literal["pending", "running", "success", "error"] = "pending"
    timestamp: str


class AgentExecutionBase(BaseModel):
    session_id: str
    agent: AgentIdentity
    status: Literal["idle", "running", "completed", "failed"] = "running"


class AgentExecutionCreate(AgentExecutionBase):
    pass


class AgentExecution(AgentExecutionBase):
    id: str
    thoughts: List[str] = []
    tool_calls: List[ToolCallRecord] = []
    result: Optional[str] = None
    started_at: datetime
    completed_at: Optional[datetime] = None
    duration_ms: Optional[int] = None

    class Config:
        from_attributes = True


class ServerMessage(BaseModel):
    type: str
    session_id: Optional[str] = None
    agent: Optional[AgentIdentity] = None
    payload: Any
    timestamp: str
    sequence: int
